src/sentry/sentry_metrics/querying/visitors.py

```diff
@@ -97,16 +117,117 @@ def _visit_timeseries(self, timeseries: Timeseries) -> QueryExpression:
         return timeseries
 
 
+class TimeseriesConditionInjectionVisitor(QueryExpressionVisitor[QueryExpression]):
+    """
+    Visitor that recursively injects a `ConditionGroup` into all `Timeseries`.
+    """
+
+    def __init__(self, condition_group: ConditionGroup):
+        self._condition_group = condition_group
+
+    def _visit_timeseries(self, timeseries: Timeseries) -> QueryExpression:
+        if self._condition_group:
+            current_filters = timeseries.filters if timeseries.filters else []
+            current_filters.extend(self._condition_group)
+
+            return timeseries.set_filters(current_filters)
+
+        return timeseries
+
+
 class ValidationVisitor(QueryExpressionVisitor[QueryExpression]):
     """
-    Visitor that recursively validates the query expression.
+    Visitor that recursively validates the `QueryExpression`.
     """
 
     def _visit_timeseries(self, timeseries: Timeseries) -> QueryExpression:
         # This visitor has been kept in case we need future validations.
         return timeseries
 
 
+class ValidationV2Visitor(QueryExpressionVisitor[QueryExpression]):
```

Comment:
nit: do we plan on renaming this at some point? Or removing the `ValidationVisitor` and naming this class  `ValidationVisitor` instead?

Reply:
Yep, as soon as we switch over to the new `metrics/query` endpoint I will fade out properly all the code.